src/theme/shadow/types.ts

```diff
@@ -0,0 +1,23 @@
+import { ShadowScheme } from "../types";
+import { StyledComponentProps } from "../helpers";
+
+type ShadowValue = (props: StyledComponentProps) => string;
```

Comment:
for future proofing, could we let ShadowValue be a function that returns `(props: StyledComponentProps) => CSSProp`, similar to border utils?